codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/core/KotlinSymbolProvider.kt

```diff
@@ -237,6 +244,37 @@ class KotlinSymbolProvider(private val model: Model, private val settings: Kotli
         }
     }
 
+    private fun getDefaultValueForTimestamp(node: NumberNode): String {
+        val instant = RuntimeTypes.Core.Instant
+
+        return if (node.isFloatingPointNumber) {
+            val fromEpochMilliseconds = RuntimeTypes.Core.fromEpochMilliseconds
+
+            val value = node.value as Double
+            val ms = round(value * 1e3).toLong()
+            "$fromEpochMilliseconds.invoke($instant, $ms)"
+        } else {
+            "$instant.fromEpochSeconds(${node.value}, 0)"
+        }
```

Comment:
**Comment**: I don't recall seeing string interpolation of symbols in our codegen before. Normally we use a formatter (e.g., `#T`). I'm guessing this works but it feels odd to me.

Reply:
We don't have access to a writer here so using fully qualified symbol names is the least invasive thing available to do probably.